src/vs/workbench/browser/parts/editor/editorCommands.ts

```diff
@@ -466,6 +474,26 @@ function registerDiffEditorCommands(): void {
 		primary: undefined,
 		handler: accessor => toggleDiffIgnoreTrimWhitespace(accessor)
 	});
+
+	MenuRegistry.appendMenuItem(MenuId.CommandPalette, {
+		command: {
+			id: TOGGLE_BRACKET_PAIR_COLORIZATION_ENABLED,
+			title: {
+				value: localize('toggleBracketPairColorization', "Toggle Bracket Pair Colorization"),
+				original: 'Toggle Bracket Pair Colorization'
+			},
+			category: CATEGORIES.View,
+			when: EditorContextKeys.editorTextFocus,
+		}
+	});
+
+	KeybindingsRegistry.registerCommandAndKeybindingRule({
+		id: TOGGLE_BRACKET_PAIR_COLORIZATION_ENABLED,
+		weight: KeybindingWeight.WorkbenchContrib,
+		when: undefined,
+		primary: undefined,
+		handler: accessor => toggleBracketPairColorization(accessor)
+	});
```

Comment:
Oh hang on this doesn't look right. We definitely don't want a keybinding by default, just a command, so this should be simplified to:

```ts
CommandsRegistry.registerCommand({
  ...
})
```

No need specifying weight/when/primary as they're unused.

Reply:
`registerAction2` would probably be even better.

Reply:
> @hediet I'm also questioning whether we should have this at all as there are extensions out there that allow easily setting up toggle settings keybindings. Bracket pair colorizing seems like it would be a configure once and you're done thing, so this command feels extremely niche to me and is therefore bloat that should be filled by extensions for those that need it.

For some reason, there is this extension with about 124k downloads: https://marketplace.visualstudio.com/items?itemName=dzhavat.bracket-pair-toggler
I'd rather have this toggle command built in than people install that extension for nothing.

Reply:
There's always requests for toggling x or y. I always close them as extension candidates. I'm not saying it's not valid but maybe we want a better way to do this or else we will have a lot of toggles in core

Reply:
👍 to what @lramos15 said. There's no harm leaving this up to extensions and there's isn't something particularly special about this setting compared to the other 1k+.